api/provider/aws/deploy_create.go

```diff
@@ -0,0 +1,133 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (d *DeployProvider) Create(req models.CreateDeployRequest) (*models.Deploy, error) {
+	deployID := generateEntityID(req.DeployName)
```

Comment:
So I was thinking about this yesterday, and there's a pretty significant flaw with this pattern is it stands: it doesn't leverage aws's versioning system at all. Each deploy will get a random ID, and will always be version `1` for that family. 

Fortunately, I think there is a relatively simple fix: instead of using the `fqDeployID` as the `familyName`, pass along the fully-qualified `req.DeployName`, e.g.

```
deployID := generateEntityID(req.DeployName)
familyName := addLayer0Prefix(d.Config.Instance(), eq.DeployName)

taskDefinition, err := d.renderTaskDefinition(req.DeployFile, familyName)
...
```

This change will allow us to leverage aws's versioning system, while still maintaining our entity ID system

Reply:
@tlake: if we decide to make this changes, it'll have some effects on how we do deploy read and lists

Reply:
oddly enough, I made essentially these same changes this morning. After manual testing :p

Reply:
@jparsons04 ^^^

Reply:
I made this comment in my [documentation PR)(https://github.com/quintilesims/layer0/pull/365#discussion_r141735889), but if we are going to use `req.DeployName`, then I feel that [CreateTaskRequest](https://github.com/quintilesims/layer0/blob/api-refactor/common/models/create_task_request.go) also needs a notion of `DeployName`

Reply:
✅

Reply:
Disregard my previous statement. I don't think we're going to add `DeployName` to CreateTaskRequest.